File: backend/app/routers/notifications.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.core.db import get_db
from app.core.deps import get_current_user
from app.models.notifications import Notification
from app.models.user import User
from app.schemas.notification import NotificationOut

router = APIRouter(prefix="/notifications", tags=["notifications"])


@router.get("", response_model=list[NotificationOut])
def list_my_notifications(
    db: Session = Depends(get_db),
    user: User = Depends(get_current_user),
) -> list[NotificationOut]:
    """The signed-in user's own notifications, newest first.

    Scoped by row (`user_id == caller`), same principle as the assigned-events
    routes: everyone can hold this endpoint, but only your own rows come back.
    """
    notifications = (
        db.query(Notification)
        .filter(Notification.user_id == user.id)
        .order_by(Notification.created_at.desc(), Notification.id.desc())
        .all()
    )
    return [NotificationOut.model_validate(n) for n in notifications]

File: frontend/src/lib/notifications.ts
// Mirror of backend/app/routers/notifications.py and schemas/notification.py.

import { apiFetch } from './api'

export interface Notification {
  id: number
  event_id: number | null
  type: string
  message: string
  is_read: boolean
  created_at: string
}

/** The signed-in user's own notifications, newest first. Scoped
 *  server-side to `user_id == me` -- there is no "whose?" parameter. */
export function listMyNotifications(): Promise<Notification[]> {
  return apiFetch('/notifications') as Promise<Notification[]>
}
